src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c

```diff
@@ -32,37 +34,327 @@ const ActorInit Obj_Ice_Poly_InitVars = {
     (ActorFunc)ObjIcePoly_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80932320 = {
-    { COLTYPE_HARD, AT_ON | AT_TYPE_ENEMY, AC_ON | AC_HARD | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0xF7CFFFFF, 0x02, 0x00 }, { 0xF7CFF7FF, 0x00, 0x00 }, TOUCH_ON | TOUCH_SFX_NONE, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit1 = {
+    {
+        COLTYPE_HARD,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_HARD | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0xF7CFFFFF, 0x02, 0x00 },
+        { 0xF7CFF7FF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NONE,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 50, 105, 0, { 0, 0, 0 } },
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_8093234C = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER | AC_TYPE_OTHER, OC1_NONE, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK5, { 0xF7CFFFFF, 0x00, 0x00 }, { 0x00000800, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sCylinderInit2 = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER | AC_TYPE_OTHER,
+        OC1_NONE,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK5,
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        { 0x00000800, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 65, 105, 0, { 0, 0, 0 } },
 };
 
-#endif
+static Color_RGBA8 D_80932378 = { 250, 250, 250, 255 };
+static Color_RGBA8 D_8093237C = { 180, 180, 180, 255 };
 
-extern ColliderCylinderInit D_80932320;
-extern ColliderCylinderInit D_8093234C;
+void ObjIcePoly_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjIcePoly* this = THIS;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Ice_Poly/ObjIcePoly_Init.s")
+    this->unk_149 = OBJICEPOLY_GET_FF00(thisx);
+    thisx->params &= 0xFF;
+    thisx->uncullZoneForward = 5600.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Ice_Poly/ObjIcePoly_Destroy.s")
+    Actor_SetScale(thisx, thisx->params * 0.01f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Ice_Poly/func_80931828.s")
+    for (i = 0; i < ARRAY_COUNT(this->colliders1); i++) {
+        Collider_InitAndSetCylinder(globalCtx, &this->colliders1[i], thisx, &sCylinderInit1);
+        Collider_InitAndSetCylinder(globalCtx, &this->colliders2[i], thisx, &sCylinderInit2);
```

Comment:
these have to be thisx to match? interesting

Reply:
I don't think so, but thisx was needed in the function for the collider assignments, and from what I know, if you need to use thisx, you need to use it as much as possible. I can change those 2 back, but that wasn't what I heard to do.